test/conditionSpec.js

```
@@ -220,16 +252,91 @@ describe('Plugin transfers (universal)', function () {
         expiresAt: makeExpiry(timeout)
       }, transferA))
 
-      try {
-        yield this.pluginA.fulfillCondition(fakeId, fulfillment)
-      } catch (err) {}
+      yield this.pluginA.fulfillCondition(fakeId, fulfillment)
+        .catch((e) => {
+          assert.equal(e.name, 'TransferNotFoundError')          
+        })
 
       yield promise
 
       sinon.assert.notCalled(fulfillStub)
     })
   })
 
+  describe('getFulfillment', () => {
+    const condition = 'cc:0:3:47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU:0'
+    const fulfillment = 'cf:0:'
+
+    it('should get the fulfillment of a completed transfer', function (done) {
+      const id = uuid()
+
+      this.pluginB.once('incoming_prepare', (transfer) => {
+        assert.equal(transfer.id, id)
+        assert.equal(transfer.ledger, 'test.nerd.')
+
+        this.pluginA.once('outgoing_fulfill', (transfer) => {
+          assert.equal(transfer.id, id)
+          assert.equal(transfer.ledger, 'test.nerd.')
+
+          this.pluginA.getFulfillment(transfer.id)
+            .then((f) => {
+              assert.equal(f, fulfillment)
+              done()
+            })
+        })
+
+        this.pluginB.fulfillCondition(id, fulfillment)
+          .then((result) => {
+            assert.isNull(result, 'fulfillCondition should resolve to null')
```

I think if this assert fails the error will be swallowed